File: robot/Comm/Messages/Remote/FromRobot/TeamColorMessage.cpp
#include <string>

#include "TeamColorMessage.h"

TeamColorMessage::TeamColorMessage(bool isBlue)
  : RemoteMessageFromRobot(RemoteMessageFromRobot::teamColor, 0, NULL) {
  createData(isBlue);
}

TeamColorMessage::~TeamColorMessage() {
}

void TeamColorMessage::createData(bool isBlue) {
  data = new char[1];
  if (data == NULL) {
    size = 0;
    return;
  }

  data[0] = (isBlue ? 1 : 0);
  size = 1;
}

File: tools/Visualizer/src/network/message/fromrobot/TeamColorMessage.java
package network.message.fromrobot;

import network.message.RemoteMessageFromRobot;

public class TeamColorMessage extends RemoteMessageFromRobot {

  public TeamColorMessage(boolean isBlue) {
    super(RemoteMessageFromRobot.MessageType.TEAM_COLOR, null);
    
    this.isBlue = isBlue;
  }
  
  public boolean isBlue() {
    return isBlue;
  }
  
  private boolean isBlue;

  public static TeamColorMessage create(byte [] data) {
    try {
      // The first byte is the boolean
      boolean isBlue = (data[0] != 0);
      
      return new TeamColorMessage(isBlue);
    }
    catch (Exception e) {
      return null;
    }
  }
}
